internal/resp/value/value.go

```diff
@@ -110,26 +112,29 @@ func (v *Value) FormattedString() string {
 	case BULK_STRING:
 		return fmt.Sprintf("%q", v.String())
 	case ARRAY:
-		formattedStrings := make([]string, len(v.Array()))
-
-		for i, value := range v.Array() {
-			formattedStrings[i] = value.FormattedString()
+		interfaceArray := make([]interface{}, len(v.array))
+		for i, val := range v.Array() {
+			interfaceArray[i] = val.ToSerializable()
 		}
-
-		return fmt.Sprintf("[%v]", strings.Join(formattedStrings, ", "))
+		respJson, err := json.MarshalIndent(interfaceArray, "", "  ")
```

Comment:
```suggestion
		respJson, err := json.MarshalIndent(v.ToSerializable(), "", "  ")
```

Could we just do this instead of constructing an array again above?